grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -217,22 +258,92 @@ GrpclbClientLoadRecorder getLoadRecorder() {
     return lbStream.loadRecorder;
   }
 
-  private class LbStream implements StreamObserver<LoadBalanceResponse> {
+  private void updateRoundRobinLists(
+      List<DropEntry> newDropList, List<BackendAddressGroup> newBackendAddrList,
+      @Nullable GrpclbClientLoadRecorder loadRecorder) {
+    HashMap<EquivalentAddressGroup, Subchannel> newSubchannelMap =
+        new HashMap<EquivalentAddressGroup, Subchannel>();
+    List<BackendEntry> newBackendList = new ArrayList<BackendEntry>();
+
+    for (BackendAddressGroup backendAddr : newBackendAddrList) {
+      EquivalentAddressGroup eag = backendAddr.getAddresses();
+      Subchannel subchannel = newSubchannelMap.get(eag);
+      if (subchannel == null) {
+        subchannel = subchannels.get(eag);
+        if (subchannel == null) {
+          Attributes subchannelAttrs = Attributes.newBuilder()
+              .set(STATE_INFO,
+                  new AtomicReference<ConnectivityStateInfo>(
+                      ConnectivityStateInfo.forNonError(IDLE)))
+              .build();
+          subchannel = helper.createSubchannel(eag, subchannelAttrs);
+          subchannel.requestConnection();
+        }
+        newSubchannelMap.put(eag, subchannel);
+      }
+      BackendEntry entry;
+      // Only picks with tokens are reported to LoadRecorder
+      if (backendAddr.getToken() == null) {
+        entry = new BackendEntry(subchannel);
+      } else {
+        entry = new BackendEntry(subchannel, loadRecorder, backendAddr.getToken());
+      }
+      newBackendList.add(entry);
+    }
+
+    // Close Subchannels whose addresses have been delisted
+    for (Entry<EquivalentAddressGroup, Subchannel> entry : subchannels.entrySet()) {
+      EquivalentAddressGroup eag = entry.getKey();
+      if (!newSubchannelMap.containsKey(eag)) {
+        entry.getValue().shutdown();
+      }
+    }
+
+    subchannels = Collections.unmodifiableMap(newSubchannelMap);
+    dropList = Collections.unmodifiableList(newDropList);
+    backendList = Collections.unmodifiableList(newBackendList);
+    maybeUpdatePicker();
+  }
+
+  @VisibleForTesting
+  class FallbackModeTask implements Runnable {
+    @Override
+    public void run() {
+      helper.runSerialized(new Runnable() {
+          @Override
+          public void run() {
+            fallbackTimerExpired = true;
+            maybeUseFallbackBackends();
+          }
+        });
+    }
+  }
+
+  @VisibleForTesting
+  class LoadReportingTask implements Runnable {
```

Comment:
This class can be inner class of LbStream, and in test, still accessible by using `command.getClass().getName()` instead of `command instanceof`

Reply:
I would prefer a direct reference to the class rather than its name. It's more robust and easier to navigate in IDEs.